apps/src/templates/sectionsRefresh/QuickAssignTableHocPl.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import moduleStyles from './sections-refresh.module.scss';
+import i18n from '@cdo/locale';
+import {MARKETING_AUDIENCE} from './CurriculumQuickAssign';
+import {renderRows} from './QuickAssignTableHelpers';
+
+const TABLE_COUNT = 3;
+/*
+Represents the (collection of) tables in Curriculum Quick Assign.
+They display side by side as if they were columns, but leaving them
+as independent tables makes styling them simpler.
+*/
+export default function QuickAssignTableHocPl({
+  marketingAudience,
+  courseOfferings,
+  updateCourse,
+  sectionCourse
+}) {
+  // Begins the table construction given the first table title
+  const allTables = title => {
+    return (
+      <div className={moduleStyles.flexDisplay}>
+        {!!courseOfferings[marketingAudience] &&
+          renderTable(title, retrieveTableSplit(0))}
+        {!!courseOfferings[marketingAudience] &&
+          renderTable('', retrieveTableSplit(1))}
+        {!!courseOfferings[marketingAudience] &&
+          renderTable('', retrieveTableSplit(2))}
+      </div>
+    );
+  };
+
+  // Renders each table (one table for each column in the component)
+  const renderTable = (title, rows) => {
+    return (
+      <table className={moduleStyles.table}>
+        <thead>
+          <tr className={moduleStyles.headerRow}>
+            <td className={moduleStyles.headerCell}>
+              <div>{title}</div>
+            </td>
+          </tr>
+        </thead>
+        <tbody className={moduleStyles.tableBody}>
+          {renderRows(rows, sectionCourse, updateCourse)}
+        </tbody>
+      </table>
+    );
+  };
+
+  // Compute these values once to reduce repitition in the function below.
+  const headers = Object.keys(courseOfferings[marketingAudience]);
+  const headerCount = headers.length;
+  const minHeadersPerTable = Math.floor(headerCount / TABLE_COUNT);
+  const overflowCount = headerCount % TABLE_COUNT; // If headers do not divide evenly
+
+  /*
+  Takes in an tableIndex (0, 1, 2) and returns the corresponding reconstructed object
+  of that slice of course offerings.
+  For example: if HOC has 6 headers, calling this with tableIndex 1 (the middle table)
+  will return an object containing courseOfferings for header indexes 2-3.
+  If called with 8 headers, the first two tables will be given one additional header.
+  */
+  const retrieveTableSplit = tableIndex => {
+    var offerings = {};
+    // If the headers don't evenly divide, distribute extras starting at table 0
+    var extraHeaderCount = overflowCount > tableIndex ? 1 : 0;
+    for (var i = 0; i < minHeadersPerTable + extraHeaderCount; i++) {
+      const index = tableIndex * minHeadersPerTable + i;
```

Comment:
Is this right if there is more than one overflow? I might just need to noodle on this a little bit more...

Reply:
Yes! I can give an example- if I have 8 headers and 3 tables (remainder 2), here are the stats:
overflowCount = 2. 
For tableIndex = 0: overflowCount > tableIndex is TRUE -> extraHeaderCount = 1. 
For tableIndex = 1: overflowCount > tableIndex is TRUE -> extraHeaderCount = 1.
For tableIndex = 2: overflowCount > tableIndex is FALSE -> extraHeaderCount = 0. 
Does this help?

Reply:
On second thought.... I might rework this slightly. Would it make more sense to you to read left to right with the overflow? As in- with 4 headers, you'd expect to see the first three in the first three tables, and then the fourth back in table 1? The way I was trying to 'math it', i currently have table one displaying headers 1 and 2, but I realize that might not be as intuitive.

Reply:
That makes sense to me. Given we're not organizing the headers in any special way right now (other than "Favorites" being the first one), I think that seems reasonable.

Reply:
Okay I updated this and it's now way simpler. I switched to the overflow approach where column/table 0 will have headers 1 and 4, the next will have 2 and 5, etc.

Reply:
I also added another header into the test file so that I can truly test the R=2 overflow scenario.